packages/discovery-provider/src/api/v1/playlists.py

```diff
@@ -561,3 +567,35 @@ class GetUnclaimedPlaylistId(Resource):
     def get(self):
         unclaimed_id = get_unclaimed_id("playlist")
         return success_response(unclaimed_id)
+
+
+access_info_response = make_response(
+    "access_info_response", ns, fields.Nested(album_access_info)
+)
+
+
+@ns.route("/<string:playlist_id>/access-info")
+class GetPlaylistAccessInfo(Resource):
+    @record_metrics
+    @ns.doc(
+        id="Get Playlist Access Info",
+        description="Gets information regarding the given user's access, and any details needed to unlock access",
+        params={"playlist_id": "A Playlist ID"},
+    )
+    @ns.expect(current_user_parser)
+    @ns.marshal_with(access_info_response)
+    def get(self, playlist_id: str):
+        args = current_user_parser.parse_args()
+        decoded_id = decode_with_abort(playlist_id, full_ns)
+        current_user_id = get_current_user_id(args)
```

Comment:
I'm legitimately unsure whether this should use the auth_middleware or not 😅

Reply:
I don't think it's too sensitive, fwiw, but I could see a case for making it authed